code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1348,18 +1348,30 @@
 /mob/living/carbon/proc/check_signables_state()
 	var/obj/item/bodypart/left_arm = get_bodypart(BODY_ZONE_L_ARM)
 	var/obj/item/bodypart/right_arm = get_bodypart(BODY_ZONE_R_ARM)
-	var/empty_indexes = get_empty_held_indexes()
-	var/exit_right = (!right_arm || right_arm.bodypart_disabled)
-	var/exit_left = (!left_arm || left_arm.bodypart_disabled)
-	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && (exit_left || exit_right)))//All existing hands full, can't sign
-		return SIGN_HANDS_FULL // These aren't booleans
-	if(exit_left && exit_right)//Can't sign with no arms!
-		return SIGN_ARMLESS
+	var/available_hands = length(get_empty_held_indexes()) // missing or disabled arms will count as empty
+	var/no_right_hand = (!right_arm || right_arm.bodypart_disabled)
+	var/no_left_hand = (!left_arm || left_arm.bodypart_disabled)
+
+	// subtract these from our available_hands
+	if(no_right_hand)
+		available_hands -= 1
+	if(no_left_hand)
+		available_hands -= 1
+
+	// items like slappers/zombie claws/etc. should be ignored
+	for(var/obj/item/I in held_items)
+		if(I.item_flags & HAND_ITEM)
+			available_hands += 1
```

Comment:
Single letter var please, also we can make use of ``as anything`` here?

Reply:
Applied with https://github.com/tgstation/tgstation/pull/69840/commits/2d8e788e0b607d199b88c944a0417620614e430b

The `as anything` skips our typecheck to boost performance (which we don't need for this proc) while making the readability of the code more verbose.  We really only should be applying it in code that needs to be optimized IMO.